### DISCUSSION TITLE
Need advice on Dioxus vs Flutter

### DISCUSSION THREAD

**Discussion by Elodie-Fouchard** [fzyzcjy/flutter_rust_bridge#477] [General]
Hi,

 What do you think about the [Dioxus](https://dioxuslabs.com/) project ? At the moment it uses some [Tauri](https://tauri.studio/) package (webview), but later on it intends to make it more native.

Let's say they manage to do it, do you think it will have better performance, better build size, less resources used, etc., since it will be written in Rust ?
Or will it be extremely difficult to catch up since Google is actively working on Flutter ?

**Comment by fzyzcjy** [comment]
Personally I have not heard Dioxus before, and has not used Tauri as well. But I have heard recently, after comparing tauri vs flutter, `Rustdesk`'s author choose Flutter.

> Let's say they manage to do it, do you think it will have better performance, better build size, less resources used, etc., since it will be written in Rust ?

Btw flutter also quite fast judging from my own app of >200kloc flutter code.

**Comment by Desdaemon** [comment]
If you aren't too invested in Flutter, Dioxus is also a very compelling option especially if you are familiar with React/webdev. Generally speaking, Dioxus will be better if you use a lot of Rust for business logic and don't want to pay the serialization costs, even though it and Tauri aren't quite as battle-tested as Flutter or React. A third option is using Tauri with Flutter Web (Tauri is framework agnostic), but you might want to wait for performance improvements on Flutter's side before committing.

**Reply by davidhmays** [reply]
Why would you use Flutter Web with Tauri when Flutter's main strength's are already in desktop and mobile applications? Not sure I see the use in using Flutter's weaker area (web) and converting it to mobile and desktop via Tauri?

**Reply by Desdaemon** [reply]
Fair point, I was mostly talking about a hypothetical situation regarding how Tauri and Flutter could work together, which definitely turned out to not be the case. It's been more than a year since and Tauri now has its own UI library (Dioxus) and FRB equivalent (rspc), so things are definitely looking promising for Rust-centric client-side app development.
